lib/exercises.rb

```diff
@@ -1,21 +1,50 @@
 
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n2)
+# Space Complexity: O(n)
 
 def grouped_anagrams(strings)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
+  return [] if strings.length == 0
+  result =  {}
+  strings.each do |word|
+    char = word.chars.sort.join
+    result[char] ||= []
+    result[char] << word
+  end
+  return result.values
 end
 
 # This method will return the k most common elements
 # in the case of a tie it will select the first occuring element.
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n)
 def top_k_frequent_elements(list, k)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
-end
+  return [] if list.length == 0
+  frequency = {}
+  # number as key 
+  # frequency as value
+
+  list.each do |num|
+    if frequency[num].nil?
+      frequency[num] = 1
+    else
+      frequency[num] += 1
+    end
+  end
 
+  result = []
+  
+  frequency.keys.each do |num|
```

Comment:
How do you know the keys will be sorted by most-frequent to least-frequent.  I'm not sure this actually works.